### DISCUSSION TITLE
key is of invalid type

### DISCUSSION THREAD

**Discussion by PavelXWX** [golang-jwt/jwt#306] [Q&A] [answered]
Can't find the error. the Token always comes as a complete string
![code](https://user-images.githubusercontent.com/95478061/231103903-7c7b7af1-b9f3-4fc3-aedc-a95e4917cc01.png)
![ksnip_20230411-133439](https://user-images.githubusercontent.com/95478061/231103912-91dde005-8b14-4489-b5d5-7cc0b1be35ef.png)

**Comment by PavelXWX** [comment]
![code2](https://user-images.githubusercontent.com/95478061/231109181-f6931ca0-b570-4857-893e-37983019042e.png)

forgot to comment out, but it doesn't affect the result

**Comment by oxisto** [comment, marked as answer]
You are using an asymetric signing method (ES256, which is based on ECDSA), but you are supplying a *symmetric* key (`[]byte`). You either need to use a symmetric signing method, such as HMAC or provide a `*ecdsa.PrivateKey` for ES256.

You can find more about this in our new project page (which is still under construction): https://golang-jwt.github.io/jwt/usage/create/ and https://golang-jwt.github.io/jwt/usage/signing_methods/#signing-methods-and-key-types
